<?xml version="1.0"?>
<robot name="sub_arm">
    <!-- base link jako root -->
    <link name="base_link">
        <inertial>
            <mass value="1.0"/><!-- Zmieniono z 0.0 na 1.0 -->
            <origin xyz="0.0 0.0 -0.05"
             rpy="0.0 0.0 0.0"/>
            <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1"/>
        </inertial>
        <visual>
            <origin xyz="0 0 -0.05" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.10" length="0.05"/>
            </geometry>
            <material name="base_color">
                <color rgba="0.1 0.1 0.1 1.0"/>
            </material>
        </visual>

    </link>    <!-- joint 1 - pierwszy przegub -->
    <!-- obrot wokol osi z -->
    <joint name="joint1" type="revolute">
        <parent link="base_link"/>
        <child link="link1"/>
        <origin xyz="0 0 0.0" rpy = "0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" effort="100" velocity="1.0"/>
    </joint>

    <link name="link1">
        <inertial>
            <mass value="0.5"/>
            <origin xyz="0 0 0.0" rpy="0 0 0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.005"/>
        </inertial>
        <visual>
            <origin xyz="0 0 0.0" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.02" length="0.05"/>
            </geometry>
            <material name="link1_color">
                <color rgba="0.9 0.9 0.9 1.0"/>
            </material>
        </visual>

    </link>

    <!-- joint 2 - pierwszy przegub -->
    <!-- origin wyniesiony o 0.05-->
    <!-- obrot wokol osi y -->
    <joint name="joint2" type="revolute">
        <parent link="link1"/>
        <child link="link2"/>
        <origin xyz="0 0 0.05" rpy = "0 0 0"/>
        <axis xyz="0 1 0"/>
        <limit lower="-3.14159" upper="3.14159" effort="100" velocity="1.0"/>
    </joint>

    <link name="link2">
        <inertial>
            <mass value="0.5"/>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.005"/>
        </inertial>
        <visual>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.02" length="0.05"/>    
            </geometry>
            <material name="link2_color">
                <color rgba="0.4 0.4 0.4 1.0"/>
            </material>
        </visual>
    </link>


    <!-- link 3-->
        <!-- obrot wokol osi y -->
    <joint name="joint3" type="revolute">
        <parent link="link2"/>
        <child link="link3"/>
        <origin xyz="0 0 0.05" rpy = "0 0 0"/>
        <axis xyz="0 1 0"/>
        <limit lower="-3.14159" upper="3.14159" effort="100" velocity="1.0"/>
    </joint>

    <link name="link3">
        <inertial>
            <mass value="0.5"/>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.005"/>
        </inertial>
        <visual>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.02" length="0.05"/>    
            </geometry>
            <material name="link2_color">
                <color rgba="0.9 0.9 0.9 1.0"/>
            </material>
        </visual>
    </link>

    <!-- link 4-->
        <!-- obrot wokol osi z -->
    <joint name="joint4" type="revolute">
        <parent link="link3"/>
        <child link="link4"/>
        <origin xyz="0 0 0.05" rpy = "0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" effort="100" velocity="1.0"/>
    </joint>

    <link name="link4">
        <inertial>
            <mass value="0.5"/>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.005"/>
        </inertial>
        <visual>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.02" length="0.05"/>    
            </geometry>
            <material name="link4_color">
                <color rgba="0.4 0.4 0.4 1.0"/>
            </material>
        </visual>
    </link>

        <!-- link 5-->
        <!-- obrot wokol osi y -->
    <joint name="joint5" type="revolute">
        <parent link="link4"/>
        <child link="link5"/>
        <origin xyz="0 0 0.05" rpy = "0 0 0"/>
        <axis xyz="0 1 0"/>
        <limit lower="-3.14159" upper="3.14159" effort="100" velocity="1.0"/>
    </joint>
    
    <link name="link5">
        <inertial>
            <mass value="0.5"/>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.005"/>
        </inertial>
        <visual>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.02" length="0.05"/>    
            </geometry>
            <material name="link4_color">
                <color rgba="0.9 0.9 0.9 1.0"/>
            </material>
        </visual>
    </link>

     <!-- link 6-->
        <!-- obrot wokol osi z -->
    <joint name="joint6" type="revolute">
        <parent link="link5"/>
        <child link="link6"/>
        <origin xyz="0 0 0.05" rpy = "0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" effort="100" velocity="1.0"/>
    </joint>
    
    <link name="link6">
        <inertial>
            <mass value="0.5"/>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.005"/>
        </inertial>
        <visual>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <geometry>
                <cylinder radius="0.02" length="0.05"/>    
            </geometry>
            <material name="link4_color">
                <color rgba="0.4 0.4 0.4 1.0"/>
            </material>
        </visual>
    </link>

     <!-- link 7 & 8- GRIPPER ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ -->
        <!-- obrot wokol osi y -->
    <joint name="joint7" type="fixed">
        <parent link="link6"/>
        <child link="link7"/>
        <origin xyz="0.02 0 0.05" rpy = "0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" effort="100" velocity="1.0"/>
    </joint>

    <link name="link7">
        <inertial>
            <mass value="0.5"/>
            <origin xyz="0 0 0.02" rpy="0 0 0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.005"/>
        </inertial>
        <visual>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <geometry>
                <box size="0.01 0.02 0.055"/>  
            </geometry>
            <material name="link4_color">
                <color rgba="0.1 0.1 0.2 1.0"/>
            </material>
        </visual>
    </link>

    <joint name="joint8" type="fixed">
        <parent link="link6"/>
        <child link="link8"/>
        <origin xyz="-0.02 0 0.05" rpy = "0 0 0"/>
        <axis xyz="0 0 1"/>
        <limit lower="-3.14159" upper="3.14159" effort="100" velocity="1.0"/>
    </joint>

    <link name="link8">
        <inertial>
            <mass value="0.5"/>
            <origin xyz="0 0 0.02" rpy="0 0 0"/>
            <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.005"/>
        </inertial>
        <visual>
            <origin xyz="0 0 0.025" rpy="0 0 0"/>
            <geometry>
                <box size="0.01 0.02 0.055"/>  
            </geometry>
            <material name="link4_color">
                <color rgba="0.1 0.1 0.2 1.0"/>
            </material>
        </visual>
    </link>

    <!-- GRIPPER_END~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ -->
</robot>

